### PR TITLE
php.extensions.uuid: init at v1.2.1

### PR DESCRIPTION
**Pull request by Zuruuh**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/php-packages/uuid/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  version = "v1.2.1";
+in
+buildPecl {
+  inherit version;
+  pname = "uuid";
+
+  src = fetchFromGitHub {
+    owner = "php";
+    repo = "pecl-networking-uuid";
+    rev = version;
```
Comment (drupol) [inline pkgs/development/php-packages/uuid/default.nix:17]:
```suggestion
    rev = "refs/tags/${version}";
```

File: pkgs/development/php-packages/uuid/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-C4SoSKkCTQOLKM1h47vbBgiHTG+ChocDB9tzhWfKUsw=";
+  };
+
+  buildInputs = [ libuuid ];
+  makeFlags = [ "phpincludedir=$(dev)/include" ];
+  doCheck = true;
+
+  PHP_UUID_DIR = libuuid;
```
Comment (drupol) [inline pkgs/development/php-packages/uuid/default.nix:25]:
```suggestion
  env.PHP_UUID_DIR = libuuid;
```

File: pkgs/development/php-packages/uuid/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-C4SoSKkCTQOLKM1h47vbBgiHTG+ChocDB9tzhWfKUsw=";
+  };
+
+  buildInputs = [ libuuid ];
+  makeFlags = [ "phpincludedir=$(dev)/include" ];
+  doCheck = true;
+
+  env.PHP_UUID_DIR = libuuid;
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/development/php-packages/uuid/default.nix:28]: We usually try to get rid of `with lib;` in front of attributes set.

**Comment by drupol** [comment]
@ofborg build php.extensions.uuid

**Review by drupol — APPROVED** [review]
Nickel! Merci!

**Comment by drupol** [comment]
Do you think it would be possible to fix the Darwin build? See the error log at https://github.com/NixOS/nixpkgs/pull/360352/checks?check_run_id=33731158722

**Comment by Zuruuh** [comment]
I will look into it :+1:

**Comment by Zuruuh** [comment]
After looking into it, it seems the `libuuid` package is `null` on darwin because of this line https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L12372
```nix
libuuid = if stdenv.hostPlatform.isLinux
    then util-linuxMinimal
    else null;
```

Though on `search.nixos.org` the `libuuid` package is indeed available on `aarch64-darwin` and `x86_64-darwin` https://search.nixos.org/packages?channel=24.11&type=packages&query=libuuid 🤔 honestly I'm not sure what to do. I've tried looking into the `libossp_uuid` library instead but the php extension isn't compatible with it

**Comment by drupol** [comment]
In that case, can you please adjust php.extensions.uuid's `meta.platforms` accordingly?

**Pull request merged**